### PR TITLE
Add documentation to many public members and populate the example dir

### PR DESCRIPTION
**Pull request by kevmoo**
Should increase the score of the package

Prepare to release v0.4.1

### CODE REVIEW

File: functions_framework/README.md [hunk 1]
```diff
@@ -86,7 +86,7 @@ properly format your code.
 
 ## Licensing
 
-Apache 2.0; see [`LICENSE`](LICENSE) for details.
+Apache 2.0; see [`LICENSE`][license] for details.
```
Comment (kevmoo) [inline functions_framework/README.md:89]: this is annoying, but necessary to make dartdoc happy!

**Comment by kevmoo** [comment]
Trying to fix the score: 
image

File: functions_framework/lib/functions_framework.dart [hunk 2]
```diff
@@ -12,6 +12,18 @@
...
+/// Provides the features needed to *define* Cloud Functions.
+///
+/// ```dart
+/// // lib/functions.dart
+/// import 'package:functions_framework/functions_framework.dart';
+/// import 'package:shelf/shelf.dart';
+///
+/// @CloudFunction()
+/// Response function(Request request) => Response.ok('Hello, World!');
+// ```
```
Comment (grant) [inline functions_framework/lib/functions_framework.dart:24]: Can we use // or /// consistently in this comment? The \`\`\` indentation does not align.

File: functions_framework/lib/src/cloud_event.dart [hunk 3]
```diff
@@ -16,6 +16,10 @@ import 'package:json_annotation/json_annotation.dart';
 
 part 'cloud_event.g.dart';
 
+/// Represents [Cloud Event](https://cloudevents.io/) data.
```
Comment (grant) [inline functions_framework/lib/src/cloud_event.dart:19]: CloudEvents one word. Ditto elsewhere.

File: functions_framework/example/readme.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# Function Framework Examples
```
Comment (grant) [inline functions_framework/example/readme.md:1]: # Functions Framework Examples

**Review by grant — APPROVED** [review]

File: functions_framework/lib/serve.dart [hunk 5]
```diff
@@ -12,6 +12,16 @@
...
 
+/// Provides the features needed to *execute* Cloud Functions.
+///
+/// Typically, this library is imported in `bin/server.dart` or similar.
+///
+/// While it's possible to use this library from hand-written code, you should
+/// use
+/// [package:functions_framework_builder](https://pub.dev/packages/functions_framework_builder)
+/// to generate server code instead.
+library serve;
```
Comment (subfuzion) [inline functions_framework/lib/serve.dart:23]: 👍

File: functions_framework/lib/functions_framework.dart (hunk 2 above)
Comment (subfuzion) [reply functions_framework/lib/functions_framework.dart:24]: Yeah, I think it's just the last line of the doc comment that needs to be `///`.

Comment (kevmoo) [reply functions_framework/lib/functions_framework.dart]: that's a mistake on my part!

**Review by subfuzion — APPROVED** [review]

**Comment by kevmoo** [comment]
@grant @subfuzion – ya'll should play with "suggest changes" in PRs – is nice!

**Pull request merged**